src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,30 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
```

Comment:
The condition `animal.hidden === false` ensures that only visible `Herbivore` instances can be bitten. However, if a `Herbivore` is already hidden, the `Carnivore` will not affect its health. Ensure this behavior is intended, as it may prevent `Carnivore` from interacting with hidden `Herbivore` instances.